turn_by_turn/utils.py

```diff
@@ -176,3 +176,36 @@ def numpy_to_tbt(names: np.ndarray, matrix: np.ndarray) -> TbtData:
         )
         indices.append(index)
     return TbtData(matrices=matrices, bunch_ids=indices, nturns=nturns)
+
+
+def numpy_to_sim_tbt(names: np.ndarray, matrix: np.ndarray) -> TbtData:
+    """
+    Converts turn by turn matrices and names into a ``TbTData`` object.
+
+    Args:
+        names (np.ndarray): Numpy array of BPM names.
+        matrix (np.ndarray): 4D Numpy array [quantity, BPM, particle/bunch No., turn No.]
+            quantities in order [x, px, y, py, t, pt, s, E].
+
+    Returns:
+        A ``TbtData`` object loaded with the matrices in the provided numpy arrays.
+    """
+    # get list of TbTFile from 4D matrix ...
+    _, _, nbunches, nturns = matrix.shape
+    matrices = []
+    indices = []
+    for index in range(nbunches):
+        matrices.append(
+            SimulationData(
```

Comment:
**{field: pd.DataFrame(index=names, data=matrix[idx, :, index, :]) for idx, field in enumerate(SIMDATA_FIELDS)}

Reply:
also, this function could have the datatype as input:
```
DataType = Union[TransverseData, SimulationData]

def numpy_to_tbt(names: np.ndarray, matrix: np.ndarray, data_type: DataType = TransverseData) -> TbtData:
```